script.js

```diff
@@ -0,0 +1,908 @@
+// Three of a kind         x          x
+// Two Pair                x          x
+// One Pair(J or Better)   x          x
+
+// Function that checks for 1 pairs to 5 of a kind
+const checkForKindsnPairs = () => {
+  // Refers to the max number of common cards found through each round of comparison
+  const maxNum = Math.max(...numCommonCardsArray);
+  // Filter the array only for number of times
+  // common card is 1 (i.e 1 pair) and copies into an array
+  const numPairs = numCommonCardsArray.filter((num) => num === 1);
+  // Check for 4 and 5 of a kind
+  if (maxNum === 3) {
+  // Next Check for a joker for 5 of a kind
+    if (playerHand.some((card) => card.name === 'Joker')) {
+      nameOfWinCombi = '5 of a kind!';
+      rankOfHand = 0;
+      isAnyKindOrPair = true;
+      return isAnyKindOrPair;
+    }
+    nameOfWinCombi = '4 of a kind!';
+    rankOfHand = 2;
+    isAnyKindOrPair = true;
+
+  // Check for 3 of a kind
+  } else if (maxNum === 2) {
+    // Get an array of unique cards
+    // if only 2 unique cards and 1 of the cards occured 3 times...
+    const uniqueCards = [...new Set(playerHand.map((card) => card.name))];
+    if (uniqueCards.length === 2) {
+      nameOfWinCombi = 'Full House';
+      rankOfHand = 0;
+      isFullHouse = true;
+      return isFullHouse;
+    }
+    nameOfWinCombi = '3 of a kind';
+    rankOfHand = 6;
+    isAnyKindOrPair = true;
+    return isAnyKindOrPair;
+
+    // Check for pair(s)
+  } else if (maxNum === 1) {
+    // Check if there are Jacks or better
+    const countJacksInHand = playerHand.filter((card) => card.rank === 11);
+    const countQueensInHand = playerHand.filter((card) => card.rank === 12);
+    const countKingsInHand = playerHand.filter((card) => card.rank === 13);
+    const countAcesInHand = playerHand.filter((card) => card.rank === 1);
+
+    if (numPairs.length === 2) {
+      nameOfWinCombi = '2 Pairs';
+      rankOfHand = 7;
+      isAnyKindOrPair = true;
+      return isAnyKindOrPair;
+    } if (countJacksInHand.length === 2 || countQueensInHand.length === 2
+      || countKingsInHand.length === 2 || countAcesInHand.length === 2) {
+      nameOfWinCombi = 'Pair of Jacks or better';
+      rankOfHand = 8;
+      isAnyKindOrPair = true;
+      return isAnyKindOrPair;
+    }
+  }
+};
+
+// Function that checks for a flush recursively,
+// taking in playerHand.length-1 initially
+const checkForFlush = (numCardsToCheck) => {
+  let i = numCardsToCheck;
+  if (i > 1) {
+    if (playerHand[i].suit === playerHand[i - 1].suit) {
+      i -= 1;
+      return checkForFlush(i);
+    }
+    // If any 2 consecutive cards'suits don't match then immediately break
+    // and declare that there is no flush;
+    isFlush = false;
+    return isFlush;
+  }
+  // Once we reached the base case (first 2 cards in playerHand array),
+  // check if the 2 cards are the same
+  if (playerHand[i].suit === playerHand[i - 1].suit) {
+    nameOfWinCombi = 'Flush!';
+    rankOfHand = 4;
```

Comment:
These hand ranks should be constants, instead of hardcoding the ranks in different places in your code. e.g. `RANK_OF_HAND_STRAIGHT`. This will help to prevent errors and improve readability.